src/argilla/client/apis/datasets.py

```diff
@@ -131,8 +131,8 @@ class _DatasetApiModel(BaseModel):
     class _SettingsApiModel(BaseModel):
         label_schema: Dict[str, Any]
 
-    def find_by_name(self, name: str) -> _DatasetApiModel:
-        dataset = get_dataset(self.http_client, name=name).parsed
+    def find_by_name(self, name: str, workspace: Optional[str] = None) -> _DatasetApiModel:
```

Comment:
We need to pass the workspace also for `find_by_name` method to retrieve the proper dataset